### PR TITLE
DRY the Harvests controller

### PR DESCRIPTION
**Pull request by Br3nda**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,4 +14,5 @@ Pathogen:
 custom_plan.rb
 zeus.json
 .bundle
-.idea/**
\ No newline at end of file
+.idea/**
+public/**
```
Comment (Br3nda) [inline .gitignore]: oops. didn't mean to check this in.

but can it stay?

File: spec/controllers/harvests_controller_spec.rb [hunk 2]
```diff
@@ -120,11 +120,11 @@ def valid_attributes
 
       it "re-renders the 'new' template" do
         # Trigger the behavior that occurs when invalid params are submitted
-        Harvest.any_instance.stub(:save).and_return(false)
```
Comment (Br3nda) [inline spec/controllers/harvests_controller_spec.rb]: we don't need to stub this. the attribute on the next line will fail on their own.

File: spec/controllers/harvests_controller_spec.rb [hunk 3]
```diff
@@ -87,9 +87,9 @@ def valid_attributes
   describe "POST create" do
     describe "with valid params" do
       it "creates a new Harvest" do
-        expect {
+        expect do
           post :create, harvest: valid_attributes
-        }.to change(Harvest, :count).by(1)
+        end.to change(Harvest, :count).by(1)
```
Comment (Br3nda) [inline spec/controllers/harvests_controller_spec.rb]: rubocop clean up

**Comment by Br3nda** [comment]
```
Failures:
  1) signin don't redirect to devise pages after signin
     Failure/Error: expect(current_path).to eq root_path
       expected: "/"
            got: "/crops"
       (compared using ==)
     # ./spec/features/signin_spec.rb:33:in `block (2 levels) in <top (required)>'
```

i've restarted the job on travis

**Comment by Br3nda** [comment]
image

File: .gitignore (hunk 1 above)
Comment (pozorvlak) [inline .gitignore]: Yes, I think these should stay. `public` is mostly auto-generated, and `.idea` is temporary files for your IDE, right?

Comment (pozorvlak) [inline .gitignore]: +1 to leaving it in.

Comment (Br3nda) [inline .gitignore]: Idea was already there. I added a new line and added public

**Pull request merged**